lib/controllers/charts/mark.ts

```diff
@@ -1,32 +1,21 @@
 import { Price, Token } from '@uniswap/sdk-core';
 import { ethers } from 'ethers';
 import { Q192 } from 'lib/constants';
-import { Token as UniSubgraphToken } from 'types/generated/graphql/uniswapSubgraph';
 
 export function price(
   sqrtPriceX96: number,
-  baseCurrency: UniSubgraphToken,
-  quoteCurrency: UniSubgraphToken,
-  token0: UniSubgraphToken,
+  baseCurrency: Token,
+  quoteCurrency: Token,
+  token0: Token,
 ): Price<Token, Token> {
   return new Price(
-    subgraphTokenToToken(baseCurrency),
-    subgraphTokenToToken(quoteCurrency),
-    token0.id !== quoteCurrency.id
+    baseCurrency,
+    quoteCurrency,
+    token0.address !== quoteCurrency.address
       ? Q192.toString()
       : ethers.BigNumber.from(sqrtPriceX96).mul(sqrtPriceX96).toString(),
-    token0.id === quoteCurrency.id
+    token0.address === quoteCurrency.address
       ? Q192.toString()
       : ethers.BigNumber.from(sqrtPriceX96).mul(sqrtPriceX96).toString(),
   );
 }
-
-function subgraphTokenToToken(token: UniSubgraphToken): Token {
-  return new Token(
-    1,
```

Comment:
refactored this out into its own helper file for others to use, and ensured it took a chainId